mattermost/mattermost-webapp
# MM-14196 Add virtualised list for posts

#### Summary

**Known issues**

- [x] P1: First team switch does not reliably work because of channel sync issue.
- [x] slow network fast posting causes scroll pops upwards
- [x] text box expansion should not overlap post list content
- [ ] compact view last post jumping (due to hover menu)

**QA testing cases**

Performance with lots of data:
- busy channel with many incoming posts - verify no white space when scrolling or sitting at the bottom of the channel
- Scroll near the top of a channel with 5000+ messages and then try to switch channels. Examine perf
- Post a message when scrolled near the top of a channel with 5000+ messages
- Click "new messages below" indicator at the top of a channel with lots of posts.

Worst case scenarios:
- Editing posts and adding reactions to most recent post and posts up in the channel. Verify no flickering or scroll pop
- Channels with images, message attchements, webhook message attachments, markdown, link previews and youtube previews, long posts. Verify no pop during post, scrolling or refresh

Scrolling smoothness:
- Scroll quickly in one direction then quickly switch scrolling direction.
- Permalinks to posts in channels that aren't loaded yet
- test various browsers
- various network speeds

Misc:
- Permalinks
- Jump to message from search
- channel and team switching. Compare to community

____

Completed:

- [x] Link preview and SVG can throw off scroll correction a little. That will be addressed with https://github.com/mattermost/mattermost-webapp/pull/2334 and https://github.com/mattermost/mattermost-webapp/pull/2469
- [x] youtube previews cause LARGE pop
- [x] removing posts can cause blank space at the top of the channel
- [x] channel header missing on channels with few posts
- [x] In progress: Fix issue of flicker when a a new reaction is added (Does not happen always most likely related to the way style cache is busted).
- [x] scroll channel down to bottom when current user make a new post
- [x] Permalink view should move post to middle of screen and highlight
- [x] Remove padding below channel header
- [x] logic tweak for bottom of the channel so we never see unmounted posts (now shows 10 max)
- [x] Fix jumping to permalink when posts are not loaded yet
- [x] Scroll to bottom when adding reaction or editing most recent post
- [x] Youtube previews causing minor incorrect scroll correction
- [x] Don't scroll center channel to bottom when posting in the RHS
- [x] Loader can get stuck at the top of the screen if we can repro
- [x] minor scroll bar adjustment when entering the channel on subsequent visits
- [x] increase number of posts to 100 (in scrolling direction)- 50 (in opposite direction)
- [x] increase load threshold to 1000px
- [x] consecutive posts are being separated in center channel
- [x] stop loading two pages on first load (only load 30 posts)
- [x] deleting system messages leaves blank space
- [x] fix overlapping issue when making every alternate posts in a channel that has a lot of posts loaded
- [x] Manual loadtest of posts to verify this issue os only related to the /test command: Fix loading indicator stuck and showing blank posts above (see image below)

____


#### Ticket Link
[MM-14196](https://mattermost.atlassian.net/browse/MM-14196)
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Touches critical sections of the codebase (auth, posting, etc.)
